The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8366998725/job/22908450416) [(plain)](https://github.com/rust-lang/rust/commit/43d412465c536560ae948af299876f12aaa29223/checks/22908450416/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=zachs18
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_8ec6d039-804e-48e4-aced-04e0414eb432
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=nonnull-slice-is_empty
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_8ec6d039-804e-48e4-aced-04e0414eb432
GITHUB_REF=refs/pull/122800/merge
GITHUB_REF_NAME=122800/merge
GITHUB_REF_PROTECTED=false
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=908ea324e59e7a9ca6d26f04ac7b25218d2f15b0
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_8ec6d039-804e-48e4-aced-04e0414eb432
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_8ec6d039-804e-48e4-aced-04e0414eb432
GITHUB_TRIGGERING_ACTOR=zachs18
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/122800/merge
GITHUB_WORKFLOW_SHA=908ea324e59e7a9ca6d26f04ac7b25218d2f15b0
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_20_X64=/opt/hostedtoolcache/go/1.20.14/x64
---
..i........................i........................i................................... 4576/4635
...........................................................

failures:
error: doctest failed, to rerun pass `-p core --doc`
---- library/core/src/ptr/non_null.rs - ptr::non_null::NonNull<[T]>::is_empty (line 1582) stdout ----
error: unexpected end of macro invocation
##[error]  --> library/core/src/ptr/non_null.rs:1586:29
Build completed unsuccessfully in 0:04:43
Build completed unsuccessfully in 0:04:43
   |
7  | assert_eq!(!slice.is_empty());
   |                             ^ missing tokens in macro arguments
note: while trying to match `,`
  --> /checkout/library/core/src/macros/mod.rs:37:16
   |
   |
37 |     ($left:expr, $right:expr $(,)?) => {

error: aborting due to 1 previous error

Couldn't compile the test.
```

</details>

--- a/library/core/src/ptr/non_null.rs
+++ b/library/core/src/ptr/non_null.rs
@@ -1580,16 +1580,14 @@ impl<T> NonNull<[T]> {
     /// # Examples
     ///
     /// ```rust
+    /// #![feature(slice_ptr_is_empty_nonnull)]
     /// use std::ptr::NonNull;
     ///
     /// let slice: NonNull<[i8]> = NonNull::slice_from_raw_parts(NonNull::dangling(), 3);
     /// assert!(!slice.is_empty());
     /// ```
-    #[stable(feature = "slice_ptr_is_empty_nonnull", since = "CURRENT_RUSTC_VERSION")]
-    #[rustc_const_stable(
-        feature = "const_slice_ptr_is_empty_nonnull",
-        since = "CURRENT_RUSTC_VERSION"
-    )]
+    #[unstable(feature = "slice_ptr_is_empty_nonnull", issue = "71146")]
+    #[rustc_const_unstable(feature = "const_slice_ptr_is_empty_nonnull", issue = "71146")]
     #[must_use]
     #[inline]
     pub const fn is_empty(self) -> bool {

--- a/library/core/src/ptr/non_null.rs
+++ b/library/core/src/ptr/non_null.rs
@@ -1575,6 +1575,27 @@ impl<T> NonNull<[T]> {
         self.as_ptr().len()
     }
 
+    /// Returns `true` if the non-null raw slice has a length of 0.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use std::ptr::NonNull;
+    ///
+    /// let slice: NonNull<[i8]> = NonNull::slice_from_raw_parts(NonNull::dangling(), 3);
+    /// assert!(!slice.is_empty());
+    /// ```
+    #[stable(feature = "slice_ptr_is_empty_nonnull", since = "CURRENT_RUSTC_VERSION")]
+    #[rustc_const_stable(
+        feature = "const_slice_ptr_is_empty_nonnull",
+        since = "CURRENT_RUSTC_VERSION"
+    )]
+    #[must_use]
+    #[inline]
+    pub const fn is_empty(self) -> bool {
+        self.len() == 0
+    }
+
     /// Returns a non-null pointer to the slice's buffer.
     ///
     /// # Examples

